### ISSUE TITLE
Matrix multiplication instead of element-wise multiplication in raw_to_polyvector

### ISSUE THREAD

**Issue by segaviv** [avaxman/libdirectional#69]
Hi,

In [raw_to_polyvector](https://github.com/avaxman/Directional/blob/593e397e3d6ceedcf7065b555003be3c6660d203/include/directional/raw_to_polyvector.h#L41) with sign symmetry, I think that line 41 does a matrix multiplication that fails an assertion in Debug due to incompatible dimensions and leads to some undefined behavior in a Release build.

I think the line can be changed to something like actualRoots = actualRoots.array().square() to calculate the element-wise square of the roots (if I understand the logic correctly...).

Thanks in advance for looking into it,
Aviv

**Comment by segaviv** [comment]
I think that there might also be an issue with the calculation of the polyvector coefficients later in this function:

1. I think that the "root polynomials" should be initialized to  -actualRoots.col(i) (since they correspond to (z - u_i)) instead of  actualRoots.col(i) in [lines 52 & 54](https://github.com/avaxman/Directional/blob/593e397e3d6ceedcf7065b555003be3c6660d203/include/directional/raw_to_polyvector.h#L52-L54).

2. I think the code for multiplying polynomials doesn't consider the biggets power of the polynomials (which is omitted from the representation since the coefficients is always 1).
I think adding something like:

for (int i =0; i < p1.cols(); i++) {
  newp.col(i + p2.cols()) += p1.col(i);
}
for (int i =0; i < p2.cols(); i++) {
  newp.col(i + p1.cols()) += p2.col(i);
}

after the nested loop [here](https://github.com/avaxman/Directional/blob/593e397e3d6ceedcf7065b555003be3c6660d203/include/directional/polyvector_to_raw.h#L42) could solve it.

**Comment by avaxman** [comment]
Yep both comments are true, thanks. It currently doesn't reproduce a polyvector_to_raw(raw_to_polyvector()) test.

Honestly, I don't think this function is actually being used within Directional tutorial. I think I translated it from another project, but I see the code there is very different from this one, so I might have stopped before finishing the entire work. It also doesn't have the new Directional class wrapper of directional::CartesianField like polyvector_to_raw() does, so I'm guessing I just haven't been there yet.

Do you want to try and fix it and submit a pull request? should test the above polyvector_to_raw(raw_to_polyvector())  to see it's an identity (up to small numerics).

**Comment by segaviv** [comment]
Yes, I'll be glad to suggest a fix.
I'll prepare a fix with the test you suggested and submit a pull request.
